controller/src/controller/db/client.py

```diff
@@ -64,18 +60,21 @@ def _merge_into_db_object(obj, obj_class):
     def _to_db_object(obj, obj_class):
```

Comment:
add a generic type so when implementing the type hint of the db class will be sorted:

```python
from typing import TypeVar, Generic

DBObjectType = TypeVar("DBObjectType")

class DBClient(Generic[DBObjectType]):
    ...
```

Then you can hint:

```suggestion
    def _to_db_object(obj, db_class: Type[DBObjectType]) -> DBObjectType:
```

Notice when inheriting, you need to specify the type: `SQLDBClient(DBClient[SQLAlchemy.Something])`

Reply:
irrelevant, not abstract anymore